lnd_test.go

```diff
@@ -3644,6 +3645,704 @@ func testMultiHopPayments(net *lntest.NetworkHarness, t *harnessTest) {
+	const pushAmt = btcutil.Amount(100000)
+	ctxb := context.Background()
+	timeout := time.Duration(time.Second * 15)
+	var networkChans []*lnrpc.ChannelPoint
+
+	// As preliminary setup, we'll create two new nodes: Carol and Dave,
+	// such that we now have a 4 ndoe, 3 channel topology. Dave will make
+	// a channel with Alice, and Carol with Dave. After this setup, the
+	// network topology should now look like:
+	//     Carol -> Dave -> Alice -> Bob
+	//
+	// First, we'll create Dave and establish a channel to Alice.
+
+	// create the topology as used in the Spider paper
+	// first, we create the nodes.
+	n1, err := net.NewNode("1", nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+	n2, err := net.NewNode("2", nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+	n3, err := net.NewNode("3", nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+	n4, err := net.NewNode("4", nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+	n5, err := net.NewNode("5", nil)
+	if err != nil {
+		t.Fatalf("unable to create new nodes: %v", err)
+	}
+
+	defer shutdownAndAssert(net, t, n1)
+	defer shutdownAndAssert(net, t, n2)
+	defer shutdownAndAssert(net, t, n3)
+	defer shutdownAndAssert(net, t, n4)
+	defer shutdownAndAssert(net, t, n5)
+
+	// connect nodes together
+	if err := net.ConnectNodes(ctxb, n1, n2); err != nil {
+		t.Fatalf("unable to connect n1 to n2: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, n2, n3); err != nil {
+		t.Fatalf("unable to connect n2 to n3: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, n3, n4); err != nil {
+		t.Fatalf("unable to connect n3 to n4: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, n4, n5); err != nil {
+		t.Fatalf("unable to connect n4 to n5: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, n5, n1); err != nil {
+		t.Fatalf("unable to connect n5 to n1: %v", err)
+	}
+	if err := net.ConnectNodes(ctxb, n2, n4); err != nil {
+		t.Fatalf("unable to connect n2 to n4: %v", err)
+	}
+
+	// send nodes initial funds
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, n1)
+	if err != nil {
+		t.Fatalf("unable to send coins to n1: %v", err)
+	}
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, n2)
+	if err != nil {
+		t.Fatalf("unable to send coins to n2: %v", err)
+	}
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, n3)
+	if err != nil {
+		t.Fatalf("unable to send coins to n3: %v", err)
+	}
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, n4)
+	if err != nil {
+		t.Fatalf("unable to send coins to n4: %v", err)
+	}
+	err = net.SendCoins(ctxb, btcutil.SatoshiPerBitcoin, n5)
+	if err != nil {
+		t.Fatalf("unable to send coins to n5: %v", err)
+	}
+
+	ctxt, _ = context.WithTimeout(ctxb, timeout)
+	chanPointn1n2 := openChannelAndAssert(
+		ctxt, t, net, n1, n2,
+		lntest.OpenChannelParams{
+			Amt: chanAmt,
+			PushAmt: pushAmt
+		},
+	)
+	networkChans = append(networkChans, chanPointn1n2)
```

Comment:
Can you also add a comment for every block here as to what channel its opening and what hash its collecting? Just a line or two would be great for like n1n2 and then n2n3 and so on.